src/Dfc.CourseDirectory.WebV2/DataStore/CosmosDb/QueryHandlers/UpdateProviderUkrlpDataHandler.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Threading.Tasks;
+using Dfc.CourseDirectory.WebV2.DataStore.CosmosDb.Models;
+using Dfc.CourseDirectory.WebV2.DataStore.CosmosDb.Queries;
+using Microsoft.Azure.Documents.Client;
+using OneOf.Types;
+
+namespace Dfc.CourseDirectory.WebV2.DataStore.CosmosDb.QueryHandlers
+{
+    public class UpsertProviderUkrlpDataHandler : ICosmosDbQueryHandler<UpsertProviderUkrlpData, Success>
+    {
+        public async Task<Success> Execute(
+            DocumentClient client,
+            Configuration configuration,
+            UpsertProviderUkrlpData request)
+        {
+            var documentUri = UriFactory.CreateDocumentUri(
+                configuration.DatabaseId,
+                configuration.ProviderCollectionName,
+                request.Id.ToString());
+
+            if (request.Update)
+            {
+                var response = await client.ReadDocumentAsync<Provider>(documentUri);
+
+                var provider = response.Document;
+
+                if (provider != null)
+                {
+                    provider.ProviderName = request.ProviderName;
+                    // Only update if supplied
+                    provider.ProviderContact = request.ProviderContact.Count > 0 ? request.ProviderContact : provider.ProviderContact;
```

Comment:
Update regardless

Reply:
even if UKRP has no Contact with ContactType = "P" ?

Reply:
Yep - we only care about P type any way